modules/rostests/apitests/advapi32/RegCreateKeyExW.c

```diff
@@ -0,0 +1,158 @@
+
+// Based on code from the following:
+// https://learn.microsoft.com/en-us/windows/win32/secauthz/creating-a-security-descriptor-for-a-new-object-in-c--
```

Comment:
Hmm, it's unclear to me whether we're allowed to use this directly. Supposedly it's covered by https://learn.microsoft.com/en-us/legal/mdsa?redirectedfrom=MSDN 🤷

Reply:
This is definitely not a complete copy because they mostly used T-strings. But to not take credit for most of the concepts, I included the note to give credit. They normally say somewhere that you can use their code examples, but they take no responsibility for its use.